### PR TITLE
Autonomous driving and Autonomous driving status

### LINKED ISSUE #32: Autonomous driving status
### User Story
As a medical worker, I want there to be a status bar that changes colour when autonomous driving is active, so that I know if manual controls are needed.

### Acceptance Criteria

- There should be a status bar which turns red when autonomous driving is disabled, and turns green when autonomous driving is currently activated.

### LINKED ISSUE #35: Simple autonomous driving
### User Story
As a medical worker, I want there to be an option to have the car drive autonomously, so that I do not need to constantly control it.

### Acceptance Criteria
- Manual Maneuvering screen should have a button to switch to autonomous driving.
- The user should be able to switch between autonomous driving and manual control.

### PR DESCRIPTION
**Pull request by puerto-hub**
### Changes

- Implemented the autonomous driving feature in the car.
- Added button in the manual control screen to turn autonomous driving on and off.
- Autonomous driving button changes color to green or red representing the status of the autonomous driving.

### Issues:
Resolves #32 
Resolves #35

### CODE REVIEW

**Review by JihanP — CHANGES_REQUESTED** [review]
I don't know if it's just my computer but it seems like the ino file doesn't compile in smce for me (Are you running the latest version of smce?), and one more thing we maybe want to merge the branch to master before? It seems like when I tested the app some of the newest uppdates haven't been merged

File: arduino/smartcar/smartcar.ino [hunk 1]
```diff
@@ -42,23 +46,34 @@ void setup() {
...
+  mqtt.begin("3.138.188.190", 1883, WiFi);
   //mqtt.begin("aerostun.dev", 1883, WiFi);
-  mqtt.begin(WiFi); // Will connect to localhost
+  //mqtt.begin(WiFi); // Will connect to localhost
 #else
   mqtt.begin(net);
 #endif
   if (mqtt.connect("arduino", "public", "public")) {
     mqtt.subscribe("/smartcar/control/#", 1);
     mqtt.onMessage([](String topic, String message) {
```
Comment (Marble879) [inline arduino/smartcar/smartcar.ino:57]: Similar to @JihanP, the ino file doesnt compile for me either when testing. I believe it is due to a bug relating to this line. This bug has been fixed, and the fix is on the master branch - so merging should resolve this issue.

**Review by Marble879 — CHANGES_REQUESTED** [review]

File: arduino/smartcar/smartcar.ino (hunk 1 above)
Comment (Marble879) [inline arduino/smartcar/smartcar.ino:57]: Similar to @JihanP, the ino file doesnt compile for me either when testing. I believe it is due to a bug relating to this line. This bug has been fixed, and the fix is on the master branch - so merging should resolve this issue.

**Review by Marble879 — CHANGES_REQUESTED** [review]

File: arduino/smartcar/smartcar.ino (hunk 1 above)
Comment (Marble879) [inline arduino/smartcar/smartcar.ino]: Similar to @JihanP, the ino file doesnt compile for me either when testing on SMCE 1.2.1. I believe it is due to a bug relating to this line (line 57) that has been mentioned on the SMCE error-log discord channel. This bug has been fixed, and the fix is on the master branch - so merging should resolve this issue.

**Review by Marble879 — CHANGES_REQUESTED** [review]

File: arduino/smartcar/smartcar.ino (hunk 1 above)
Comment (puerto-hub) [reply arduino/smartcar/smartcar.ino]: Ok, so should I commit a new change on line 57 or should I try to fix it when I resolve conflicts when the branch is merged?

Comment (Marble879) [reply arduino/smartcar/smartcar.ino]: I think you just need to merge the master branch into this one, and fix the conflicts. The bug fix is already on the master branch, so merging will resolve the issue and there is no need to apply a fix.

File: android/app/src/main/res/layout/activity_manual_control.xml [hunk 2]
```diff
@@ -109,4 +109,20 @@
         app:layout_constraintTop_toBottomOf="@+id/joystickView2"
         app:layout_constraintVertical_bias="0.428" />
 
+    <Button
+        android:id="@+id/button2"
```
Comment (Marble879) [inline android/app/src/main/res/layout/activity_manual_control.xml]: Renaming the button ID to something like autonomousDrivingButton for example, may improve readability/maintainability of the code.

**Review by Marble879 — CHANGES_REQUESTED** [review]
I think everything looks great! After testing, I think it would also be good for the car to check the right side after it detects an obstacle on the left. 

This may help when the car is moving down a hospital corridor and detects an obstacle. If the car would only turn left when an obstacle is detected, it would go back the way it came from. By adding a right check also, the car could continue to travel through the hospital corridor, requiring less input from the user.

**Review by Marble879 — APPROVED** [review]
Everything worked well when testing and the code also looks good! Great job :+1: 

NOTE: The modification mentioned in my previous code review, was moved to Sprint 4 (hence it is not part of this PR anymore). Details can be found at: #59

**Review by MarkusJ299 — APPROVED** [review]
Now that we moved the car checking both directions upon obstacle detection to the next sprint, I think this looks good.

**Review by NilsDunlop — APPROVED** [review]
This is a great start to the autonomous driving. We can perfect this mode in the next sprint. Really well done!

**Review by JihanP — APPROVED** [review]
Good job, this is a good base to stand on when trying to make a more advanced automated driving in next spring.

**Review by Kevinayad — APPROVED** [review]
Good job!

**Review by gusasino — COMMENTED** [review]
Looks good!

**Pull request merged**